### PR TITLE
Issue #11736: Update doc for MissingJavadocTypeCheck

### PR DESCRIPTION
**Pull request by stoyanK7**
Addresses [#11736](https://github.com/checkstyle/checkstyle/issues/11736#issuecomment-1181167741)
Updates documentation and provides one more example.

https://checkstyle.sourceforge.io/config_javadoc.html#MissingJavadocType

Screenshot from 2022-07-12 17-09-07
Screenshot from 2022-07-12 18-24-33

### CODE REVIEW

**Comment by romani** [comment]
GitHub, generate website

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java [hunk 1]
```diff
@@ -53,7 +53,8 @@
  * </li>
  * <li>
  * Property {@code skipAnnotations} - specify annotations that allow missed
- * documentation. Only short names are allowed, e.g. {@code Generated}.
+ * documentation. Both short(e.g. {@code Generated}) and qualified(e.g. {@code MyClass.Annotation})
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java:56]: We do not fully support qualified names.

We simply match name as is. If user use two ways of same annotation (short name and with  firs package) he needs to use both forms in config if do not want violation.

Not sure how to word it properly.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java (hunk 1 above)
Comment (stoyanK7) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java:56]: I think simply say as things are.?

> ..missed documentation. Specified annotations are matched as is - `Annotation` will only match `@Annotation` but not `@MyClass.Annotation`.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java:56]: `...But not any qualified form of it`

Reminder:Fully qualified name is name with package.

Comment (stoyanK7) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java:56]: That sounds better. Just to make sure before I make the changes.

>..missed documentation. Specified annotations are matched as is - Annotation will only match `@Annotation` but not any qualified form of it(e.g. `@MyClass.Annotation`).

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java [hunk 2]
```diff
@@ -127,21 +128,25 @@
...
  * </p>
  * <pre>
  * &#64;SpringBootApplication // no violations about missing comment on class
  * public class Application {}
  *
  * &#64;Configuration // no violations about missing comment on class
  * class DatabaseConfiguration {}
+ *
+ * &#64;MyClass.Annotation // no violations about missing comment on class
+ * class SomeClass {}
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java:141]: this example intention is to be real life example.

Lets move your example to separate cong+code block to show nuance of match.
```
@Annotation // ok
@MyClass.Annotation // ok
com.mycompany.MyClass.Annotation // violation, as this form is missed in config
```

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java (hunk 1 above)
Comment (stoyanK7) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java]: Done!

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java (hunk 2 above)
Comment (stoyanK7) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java]: I need some help. How do I get rid of this CI failure? I understand it is complaining that I pushed a commit with changes not staged for commit. I tried `git commit --amend --allow-empty` but no success:
```
Please clean up or update .gitattributes file.
Git status output:
```

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java]: I am not sure how you came to this.
I recommend to create new branch, reply patch or cherry pick or ..., remove on your local old branch, rename new branch to old branch, push with force to GitHub.

Comment (stoyanK7) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java]: It seems that running `mvn clean verify` modifies the file:
Screenshot from 2022-07-13 19-24-22
Screenshot from 2022-07-13 19-36-36

Is this a bug or a feature where `-`(dash) is not allowed in `.xml` files? I see other `.xml` containing such.

Comment (stoyanK7) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java]: I will try to escape it with `&#8211;` and see what happens.

Comment (stoyanK7) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java]: Same happens with `&#8211;`. `mvn clean verify` deletes it. I will just remove it.

Comment (stoyanK7) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java]: Done!

**Comment by stoyanK7** [comment]
Github, generate site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java [hunk 3]
```diff
@@ -145,6 +146,28 @@
  * &lt;/module&gt;
  * </pre>
  * <p>
+ * Example that allows missing comments for classes annotated with {@code @Annotation}
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java:149]: I suggest to change order a bit. It looks like at some previous PR usual order had changed - config should go before example code, that is a common pattern, e.g. https://checkstyle.org/config_modifier.html#InterfaceMemberImpliedModifier_Examples

Previous example (I suppose you took it as reference) can be corrected as well.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java:149]: @stoyanK7 , please do us a favor and swap config and java code blocks for both examples.

Comment (stoyanK7) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java]: Done!
Screenshot from 2022-07-14 09-42-55

**Comment by strkkk** [comment]
Github, generate site

**Review by romani — APPROVED** [review]
ok to merge

**Review by strkkk — APPROVED** [review]

**Pull request merged**